spinalcordtoolbox/modality_prediction/core.py

```diff
@@ -0,0 +1,118 @@
+        """
+        This method standardizes each slices individually
+        """
+        for i in range(len(self.slices)):
+            mean, std = self.slices[i].mean(), self.slices[i].std() 
+            self.slices[i] = (self.slices[i] - mean) / std
+    
+    def CenterCropTransform(self, size=128):
+        """
+        This method centers the image around the center
+        """
+        for i in range(len(self.slices)):
+            y, x = self.slices[i].shape
+            
+            startx = x // 2 - (size // 2)
+            starty = y // 2 - (size // 2)
+            
+            if startx < 0 or starty < 0:
+                raise RuntimeError("Negative crop.")
+            
+            self.slices[i] = self.slices[i][starty:starty + size,
+                                           startx:startx + size]
+    
+    def ToTensor(self):
+        """
+        This method returns the tensor in the correct shape to feed the network
+        ie. torch.Size([16, 1, 128, 128]) with dtype = float
+        """        
+        slices = np.asarray(self.slices, dtype=np.float32)
+        slices = np.expand_dims(slices, axis=1)
+        tensor = torch.FloatTensor(slices)
+        return(tensor)
+
+
+def classify_acquisition(input_image, model=None):
+    """
+    This function takes as input an object from spinalcordtoolbox.image object and the
+    model and outputs the predicted modality.
+
+    :param input_image: the loaded and correctly oriented Image object
+    :param model: the loaded pre-trained model
+    :return (string): name of the class
+    """
+
+    # We preprocess the slices within our Acquisition Class
+    acq = Acquisition()
+    acq.loadFromImage(input_image)
+    acq.CenterCropTransform()
+    acq.StandardizeTransform()
+    
+    input_slices = acq.ToTensor()
+
+    with torch.no_grad():
+        
+        input_slices = input_slices
+
+        outputs = model(input_slices)   
+        _, preds = torch.max(outputs, 1)
+        preds = preds.tolist()
+
+    # We consider the mode of the predictions for each slice
+    numeral=[[preds.count(nb), nb] for nb in preds]
+    numeral.sort(key=lambda x:x[0], reverse=True)
+    modality = numeral[0][1]
+
+    class_names = ["t1", "t2s", "t2"]
+    return(class_names[modality])
+
+
+def classify_from_path(input_path):
+    """
+    This is our main function that will be called from the sct scripts inside the parser.
+    :param input_path: raw path to the nifti acquisition
+    :return: the predicted modality
+    """
+
+    # We load the acquisitions from the image module in order to benefit from all existing methods
+    input_image = Image(input_path)
+    input_image.change_orientation('RPI')
+
+    # We load the model
+    model = M.Classifier()
+    model_path = os.path.join(sct.__sct_dir__, 'data', 'modality_pred_model.pt')
+    model.load_state_dict(torch.load(model_path, map_location='cpu'))
+    model.eval()
+
+    modality = classify_acquisition(input_image, model)
```

Comment:
I would recommend to print the result of the automatic modality detection at this point.
Could be very helpful for the debugging etc.
Something like:
```
logger.info('Modality detected: {}'.format(modality))
```